sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuiteBase.scala

```diff
@@ -40,6 +40,38 @@ trait ShowTablesSuiteBase extends QueryTest with DDLCommandTestUtils {
     checkAnswer(df, expected)
   }
 
+  // the error class & error parameters of
+  // `SHOW TABLE EXTENDED ... PARTITION ... in non-partitioned table`
+  protected def extendedPartInNonPartedTableError(
+      catalog: String,
+      namespace: String,
+      table: String): (String, Map[String, String])
+
+  protected def extendedPartExpectedResult: String =
+    "Partition Values: [id1=1, id2=2]"
+
+  protected def namespaceKey: String = "Database"
+
+  protected def extendedTableExpectedResultDiff: String
+
+  private def extendedTableExpectedResult(
+      catalog: String,
+      namespaceName: String,
+      namespace: String,
+      table: String,
+      partColName: String,
+      dataColName: String): String = {
+    s"""Catalog: $catalog
+       |$namespaceName: $namespace
+       |Table: $table
+       |$extendedTableExpectedResultDiff
```

Comment:
this name is a bit confusing. Do you mean `extendedTableInfo`?

Reply:
Done